clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -207,9 +207,14 @@ public synchronized void ensureCoordinatorReady() {
     }
 
     protected synchronized RequestFuture<Void> lookupCoordinator() {
-        if (findCoordinatorFuture == null)
-            findCoordinatorFuture = sendGroupCoordinatorRequest();
-        return findCoordinatorFuture;
+        RequestFuture<Void> future = findCoordinatorFuture;
+        if (future == null) {
+            future = sendGroupCoordinatorRequest();
+            // If no brokers are available, return a failed future without setting findCoordinatorFuture
```

Comment:
One minor suggestion: perhaps this would be clearer if we call `leastLoadedNode()` in this function and modify `sendGroupCoordinatorRequest` to accept the node as a parameter?